### PR TITLE
Add extra version of py-isort

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-isort/package.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class PyIsort(PythonPackage):
     homepage = "https://github.com/timothycrosley/isort"
     url      = "https://pypi.io/packages/source/i/isort/isort-4.2.15.tar.gz"
 
+    version('4.3.20', sha256='c40744b6bc5162bbb39c1257fe298b7a393861d50978b565f3ccd9cb9de0182a')
     version('4.2.15', sha256='79f46172d3a4e2e53e7016e663cc7a8b538bec525c36675fcfd2767df30b3983')
 
     depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-isort/package.py]: Missing dependencies in the latest version:
```python
      install_requires=[                                                        
          'futures; python_version < "3.2"',                                    
          'backports.functools_lru_cache; python_version < "3.2"',              
      ],                                                                        
      python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*",           
```
You'll have to see when these dependencies were added.

File: var/spack/repos/builtin/packages/py-isort/package.py [hunk 2]
```diff
@@ -12,6 +12,13 @@ class PyIsort(PythonPackage):
     homepage = "https://github.com/timothycrosley/isort"
     url      = "https://pypi.io/packages/source/i/isort/isort-4.2.15.tar.gz"
 
+    version('4.3.20', sha256='c40744b6bc5162bbb39c1257fe298b7a393861d50978b565f3ccd9cb9de0182a')
     version('4.2.15', sha256='79f46172d3a4e2e53e7016e663cc7a8b538bec525c36675fcfd2767df30b3983')
 
-    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-isort/package.py]: Are you sure this is needed at run time?

File: var/spack/repos/builtin/packages/py-isort/package.py [hunk 3]
```diff
@@ -12,6 +12,13 @@ class PyIsort(PythonPackage):
...
+    depends_on('py-pytest', type='test')
+    depends_on('py-mock', type='test')
+    depends_on('python@2.6:2.7.999,3.3:', type=('build', 'run'), when='@4.2.15:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-isort/package.py]:
```suggestion
    depends_on('python@2.6:2.8,3.3:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-isort/package.py [hunk 4]
```diff
@@ -12,6 +12,13 @@ class PyIsort(PythonPackage):
...
+    depends_on('python@2.7:2.7.999,3.4:', type=('build', 'run'), when='@4.3.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-isort/package.py]:
```suggestion
    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'), when='@4.3.0:')
```

File: var/spack/repos/builtin/packages/py-isort/package.py [hunk 5]
```diff
@@ -12,6 +12,13 @@ class PyIsort(PythonPackage):
...
+    depends_on('py-futures', type=('build', 'run'), when='@4.3.0: ^python@2.7:2.7.999,:3.1')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-isort/package.py]:
```suggestion
    depends_on('py-futures', type=('build', 'run'), when='@4.3.0: ^python@:3.1')
```

File: var/spack/repos/builtin/packages/py-isort/package.py [hunk 6]
```diff
@@ -12,6 +12,13 @@ class PyIsort(PythonPackage):
...
+    depends_on('py-backports-functools-lru-cache', type=('build', 'run'), when='@4.3.10: ^python@:3.2')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-isort/package.py]:
```suggestion
    depends_on('py-backports-functools-lru-cache', type=('build', 'run'), when='@4.3.10: ^python@:3.1')
```

File: var/spack/repos/builtin/packages/py-isort/package.py (hunk 2 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-isort/package.py]: Last question before merging this one.

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-isort/package.py]: Yes, it is needed: isort's `main.py` explicitly imports `setuptools`

**Pull request merged**